lib/Migrator/APIDiffMigratorPass.cpp

```diff
@@ -174,12 +179,14 @@ class ChildIndexFinder : public TypeReprVisitor<ChildIndexFinder, FoundResult> {
   }
 
   FoundResult visitTupleTypeRepr(TupleTypeRepr *T) {
-    // Single element TupleTypeReprs may be arbitrarily nested so don't count
-    // as their own index level
-    if (T->getNumElements() == 1) {
+    // Paren TupleTypeReprs may be arbitrarily nested so don't count as their
+    // own index level except in the case of function type argument parens
+    if (T->isParenType() && !IsFunctionTypeArgument) {
```

Comment:
To avoid situations like this in the future, I think it would be great to move the the scheme similar to AST representation of function types where instead of a single "parameter" type it holds array of parameters with their attributes.

Reply:
@xedin I think the indexing scheme should match the structure of the various TypeReprs, but flattening the single parameter type in FunctionTypeRepr sounds good to me.